diff --git a/node_modules/@nlpjs/directline-connector/src/directline-controller.js b/node_modules/@nlpjs/directline-connector/src/directline-controller.js
index 217e0d1..9ec709a 100644
--- a/node_modules/@nlpjs/directline-connector/src/directline-controller.js
+++ b/node_modules/@nlpjs/directline-connector/src/directline-controller.js
@@ -24,6 +24,8 @@
 const { uuid } = require('@nlpjs/core');
 const fetch = require('node-fetch');
 
+const noAnswer = {};
+
 class DirectlineController {
   constructor(settings, parent) {
     const localhost = 'http://localhost:3000';
@@ -193,8 +195,17 @@ class DirectlineController {
                   activity,
                 })
                 .then((nlpresult) => {
-                  result.text =
-                    nlpresult.answer || "Sorry, I didn't understand you";
+                  if (nlpresult.answer) {
+                    result.text = nlpresult.answer;
+                  } else {
+                    if (noAnswer[conversation.conversationId] && noAnswer[conversation.conversationId] >= 1) {
+                      noAnswer[conversation.conversationId] = 0;
+                      result.text = "I couldn't understand your question. Kindly contact the universirt for more information about the program";
+                    } else {
+                      noAnswer[conversation.conversationId] = 1
+                      result.text = "Sorry, I didn't find an answer for you, can you please rephrase your question";
+                    }
+                  }
                   if (activity.channelId === 'emulator') {
                     result.nlp = {
                       intent: nlpresult.intent,
